src/System.Management.Automation/DscSupport/MofCimDSCParser.cs

```diff
@@ -0,0 +1,478 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Management.Automation;
+using System.Management.Automation.Language;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Security;
+using System.Text;
+
+using Microsoft.Management.Infrastructure;
+using Microsoft.Management.Infrastructure.Generic;
+using Microsoft.Management.Infrastructure.Serialization;
+using Microsoft.PowerShell.Commands;
+
+namespace Microsoft.PowerShell.DesiredStateConfiguration.Internal
+{
+    /// <summary>
+    /// </summary>
+    [SuppressMessage("Microsoft.MSInternal", "CA903:InternalNamespaceShouldNotContainPublicTypes",
+        Justification = "Needed Internal use only")]
+    public static class DscRemoteOperationsClass
+    {
+        /// <summary>
+        /// Convert Cim Instance representing Resource desired state to Powershell Class Object.
+        /// </summary>
+        public static object ConvertCimInstanceToObject(Type targetType, CimInstance instance, string moduleName)
+        {
+            var className = instance.CimClass.CimSystemProperties.ClassName;
+            object targetObject = null;
+            string errorMessage;
+
+            using (System.Management.Automation.PowerShell powerShell = System.Management.Automation.PowerShell.Create(RunspaceMode.CurrentRunspace))
+            {
+                string script = "param($targetType,$moduleName) & (Microsoft.PowerShell.Core\\Get-Module $moduleName) { New-Object $targetType } ";
```

Comment:
@daxian-dbw might know if there's a simpler way to do this with the PSCommand/PowerShell API

Reply:
MofCimDSCParser.cs is existing code (that was moved to a new file, so it shows as green in the diff) that I'd like to avoid changing unless absolutely necessary.